nbi/engine/native/launcher/windows/src/JavaUtils.c

```diff
@@ -125,58 +125,79 @@ DWORD isJavaCompatible(JavaProperties *currentJava, JavaCompatible ** compatible
 JavaVersion * getJavaVersionFromString(char * string, DWORD * result) {
     JavaVersion *vers = NULL;
     if(getLengthA(string)>=3) {
-        //hope that at least we "major.minor" : 1.5
-        if(string[1]=='.') {
-            char c = string[0];
+        char *p = string;
+        
+        // get major 
+        long major = 0;
+        while(p!=NULL) {
```

Comment:
`p != NULL` does not terminate at the end of string. `getLengthA()` above returns 0 for `string == NULL`, so it does not need to be tested. Use `*p != 0` or just `*p` (zero termination character)